add(epp): applications-cisco-ssms-restapi EN (#329)
centreon/centreon-documentation: en/integrations/plugin-packs/procedures/applications-cisco-ssms-restapi.md

---
id: applications-cisco-ssms-restapi
title: Cisco SSMS
---

## Overview

Cisco Smart Software Manager On-Prem (formerly known as Cisco Smart Software Manager satellite) is a component of Cisco Smart Licensing
that works in conjunction with Cisco Smart Software Manager (SSM).
It offers near real-time visibility and reporting of the Cisco licensesyou purchase and consume while giving security-sensitive organizations,
a way to access a subset of Cisco SSM functionality without using a direct Internet connection to manage their install base.

## Plugin Pack Assets

### Monitored Objects

* Applications
* Servers
* Licences
* Alerts

### Collected Metrics

<!--DOCUSAURUS_CODE_TABS-->

<!--Alerts-->

| Metric Name                | Description              |
| :------------------------- | :------------------------|
| account.alerts.minor.count | Counter of alerts minor. |
| account.alerts.major.count | Counter of alerts major. |

> 

Reviewer: ```suggestion
```

<!--Licenses-->

| Metric Name                | Description                                        |

> | :------------------------- | :------------------------------------------------- |

Reviewer: Add the status

Author: Added the status

| licenses.usage.count       | Counter of licenses usage. Unit : Bytes (B)        |

> | licenses.free.count        | Counter of licenses free. Unit : Bytes (B)         |

Reviewer: Wrong unit

Author: Corrected unit

| licenses.usage.percentage  | Percentage of licenses usage. Unit : percentage(%) |

> 

Reviewer: ```suggestion
```

<!--END_DOCUSAURUS_CODE_TABS-->

> 

Reviewer: ```suggestion
```

## Prerequisites

> ### API Privileges

Reviewer: ```suggestion
```

> A service account is required to query the Cisco SSMS API. It must have sufficient read privileges in the account targeted.

Reviewer: Maybe add a reference to the documentation like this : https://www.cisco.com/c/dam/en_us/buy/smart-accounts/smart-software-manager-satellite-enhanced-edition-6-2-0-user-guide.pdf

Author: Added the reference.

> 

Reviewer: ```suggestion
```

> ## Installation

Reviewer: ```suggestion
## Setup
```

<!--DOCUSAURUS_CODE_TABS-->

<!--Online IMP Licence & IT-100 Editions-->

1. Install the Centreon Plugin on every poller monitoring Cisco SSMS resources:

```bash
yum install centreon-plugin-Applications-Cisco-Ssms-Restapi.noarch
```

2. On the Centreon Web interface in "Configuration > Plugin packs > Manager", install the *Cisco SSMS* Plugin-Pack

> 

Reviewer: ```suggestion
```

<!--Offline IMP License-->

1. Install the Centreon Plugin on every poller monitoring Cisco SSMS resources:

```bash
yum install centreon-plugin-Applications-Cisco-Ssms-Restapi.noarch
```

2. On the Centreon Central server, install the Centreon Plugin-Pack from the RPM:

```bash
yum install centreon-pack-applications-cisco-ssms-restapi.noarch
```

3. On the Centreon Web interface in "Configuration > Plugin packs > Manager", install the *Cisco SSMS* Plugin-Pack

> 

Reviewer: ```suggestion
```

<!--END_DOCUSAURUS_CODE_TABS-->

> 

Reviewer: ```suggestion
```

## Configuration

The Plugin-Pack is designed to monitor resources based on one host per account.
Adding a host into Centreon, link it to the template named *App-Cisco-Ssms-Restapi-custom*. 
Once the template applied, some Macros have to be configured:

> | Mandatory   | Nom                    | Description                                                                |

Reviewer: ```suggestion
| Mandatory   | Name                    | Description                                                                |
```

| :---------- | :--------------------- | :------------------------------------------------------------------------- |

> | X           | APIPORT                | Port used. Default is 443                                                  |

Reviewer: ```suggestion
| X           | APIPORT                | Port used. Default is 8443                                                  |
```

| X           | CLIENTID               | Client ID to access to the API.                                            |
| X           | CLIENTSECRET           | Client Secret to access to the API.                                        |
|             | APIEXTRAOPTIONS        | Any extra option you may want to add to the command                        |

## FAQ

### How to check in the CLI that the configuration is OK and what are the main options for ?

Once the Plugin installed, log into your poller using the *centreon-engine* user account and test by running the following command (Parameters such as ```account``` or ```client-id```have to be adjusted):

```bash
/usr/lib/centreon/plugins/centreon_cisco_ssms_restapi.pl \
    --plugin=apps::cisco::ssms::restapi::plugin \
	--mode=licenses \
	--hostname='myipaddress' \
	--client-id='myapiclientid' \
	--client-secret='myapiclientsecret' \
	--account='1234abc-56de-78fg-90hi-1234abcdefg' \
	--filter-counters='status' \
	--filter-license-name='mylicence'
	--critical-license-status='%{status} !~ /in compliance/i' \
	--verbose

> 	

Reviewer: ```suggestion
```

```

The command above gets the status of a Cisco SSMS licenses (```--mode=licenses```) named *mylicense* (```--filter-licence-name='mylicence'```). 
This license belongs to account *1234abc-56de-78fg-90hi-1234abcdefg* (```--account='1234abc-56de-78fg-90hi-1234abcdefg'```). 

> This command will trigger a CRITICAL alarm if it contains the word *in compliance* (```--critical-license-status='%{status} !~ /in compliance/i'``) 

Reviewer: ```suggestion
The command will trigger a CRITICAL alamer if the status of the licence is different from *in compliance* (```--critical-license-status='%{status} !~ /in compliance/i'```) 
```

All the options that can be used with this plugin can be found over the ```--help``` command:

> ```/usr/lib/centreon/plugins//centreon_cisco_ssms_restapi.pl --plugin=apps::cisco::ssms::restapi::plugin --mode=licences --help```

Reviewer: ```
/usr/lib/centreon/plugins//centreon_cisco_ssms_restapi.pl --plugin=apps::cisco::ssms::restapi::plugin --mode=licences --help
```

Author: Done!

### How do I remove the *count* perfdatas if I want to filter on just one application ?

The Plugin adds the count of objects by default. This can be useless if the objects are filtered with the ```--filter-license-name``` parameter.

> Therefore, these useless perfdatas can be omitted by adding a perfdata filter : ```--filter-perfdata='^$'```.

Reviewer: Could have something about proxy error ?
like the https://docs.centreon.com/current/en/integrations/plugin-packs/procedures/network-cisco-meraki-restapi.html#docsNav

Author: Added ending of documentation

This license belongs to account *1234abc-56de-78fg-90hi-1234abcdefg* (```--account='1234abc-56de-78fg-90hi-1234abcdefg'```). 

This command will trigger a CRITICAL alarm if the status of the license is different from *in compliance* (```--critical-license-status='%{status} !~ /in compliance/i'```) 

All the options that can be used with this plugin can be found over the ```--help``` command:

```bash
/usr/lib/centreon/plugins//centreon_cisco_ssms_restapi.pl --plugin=apps::cisco::ssms::restapi::plugin 
--mode=licences --help
```

### Why do I get the following error: 

#### ```UNKNOWN: 500 Can't connect to api.meraki.com:443```

This error message means that the Centreon Plugin couldn't successfully connect to the Cisco Meraki API.
Check that no third party device (such as a firewall) is blocking the request.
A proxy connection may also be necessary to connect to the API. This can be done by using the ```--proxyurl``` option in the command.

#### ```UNKNOWN: 501 Protocol scheme 'connect' is not supported |``` 

When using a proxy to connect to the Meraki API, this error message means that the Centreon Plugin library does not support
the proxy connection protocol.

> In order to prevent this issue, use the *curl* HTTP backend by adding the following option to the command: ```--http-backend='curl'```.

Reviewer: Remove Meraki ref from your copy/past :p

Author: Meraki ref removed ^^
